src/jit/morph.cpp

```diff
@@ -11571,7 +11571,19 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             break;
 
         case GT_DIV:
-
+            // Replace "val / dcon" with "val * (1.0 / dcon)" if dcon is a power of two.
+            // Powers of two within range are always exactly represented, 
+            // so multiplication by the reciprocal is safe in this scenario
+            if (op2->IsCnsFltOrDbl())
+            {
+                double divisor = op2->AsDblCon()->gtDconVal;
+                if ((typ == TYP_FLOAT || typ == TYP_DOUBLE) && FloatingPointUtils::isPow2(divisor))
```

Comment:
It looks like you broke the `float` case, I don't see how having a `double` only version of `isPow2` deals with the case of `float` denormal.